packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/data_type/mod.rs

```diff
@@ -87,7 +85,9 @@ impl<C: AsClient> DataTypeStore for PostgresStore<C> {
             graph_resolve_depths,
         } = *query;
 
-        let subgraphs = stream::iter(Read::<PersistedDataType>::read(self, expression).await?)
+        let mut filter = Filter::try_from(expression.clone()).change_context(QueryError)?;
```

Comment:
The plan is to get rid of `Expression`s eventually, right? I agree with Alfie that we can probably change the workspace API queries in one go once we have the JSON representation of the new filters. May make sense to pair on that and get it done soon

Reply:
Yes, that's the plan ASAP. Pairing on it sounds like a good idea. @benwerner01 and @Alfred-Mountfield should probably join as well.

Reply:
I don't think we need 4 people for it. It's a very small task in my mind. It's genuinely 11 occurrences of queries, and I think we only have 2/3 queries (basically queryAllLatest, or queryById). The TS surface area will be minimal, and the Rust changes shouldn't require much thought.

Reply:
It's not a big task but it's worth it that everyone working at the backend has seen the queries I think.

Reply:
Then please cc Ben and I on the review :), and we can also help review the docs for the syntax when they're ready

Reply:
Sure, will do 🙂